src/pages/home/report/ReportActionsView.js

```diff
@@ -127,10 +126,9 @@ class ReportActionsView extends React.Component {
                 this.scrollToListBottom();
             }
 
-            // When the last action changes, wait three seconds, then record the max action
-            // This will make the unread indicator go away if you receive comments in the same chat you're looking at
-            if (Visibility.isVisible()) {
-                this.timers.push(setTimeout(this.recordMaxAction, 3000));
+            // If we are adding a new action while the report is open, record the max action immediately
+            if (Visibility.isVisible() && prevProps.reportID === this.props.reportID && !this.unreadTimer) {
```

Comment:
What if we record the last action anyway. Would something bad happen?

Reply:
- `!this.unreadTimer`: If we have a non-selected chat with new messages and move to it, the timer triggers (3s). This prevents the last action from being recorded before that timer ends (so always wait those three seconds, even if we get a new message), although now that we have moved this to only trigger on new messages, we may not really care (it was more important before, since it would trigger with any updates)
- I haven't seen any issues with the last action

Reply:
This feels like a code smell to me. Probably if all the props updated at once we would not have this problem.